src/lib.rs

```diff
@@ -212,11 +219,14 @@ broker_handler!(AuthHandler, |app, req| {
         }
     );
     let client_id = try_get_param!(params, "client_id");
-    let nonce = try_get_param!(params, "nonce");
     let redirect_uri = try!(
         Url::parse(try_get_param!(params, "redirect_uri"))
             .map_err(|_| BrokerError::Input("redirect_uri is not a valid URL".to_string()))
     );
+    // Per the OAuth2 spec, we may redirect to the RP once we have validated client_id and
+    // redirect_uri. In our case, this means we make redirect_uri available to error handling.
+    req.extensions.insert::<RedirectUri>(redirect_uri.clone());
+    let nonce = try_get_param!(params, "nonce");
```

Comment:
Nit: can this last line not move in this commit, please? And if it moves, it should probably go after the `let email_addr`, as well? Then, maybe add a blank line before it.